### DISCUSSION TITLE
Next js node_modules tree shaking not working with monorepo

### DISCUSSION THREAD

**Discussion by VivekNeel** [vercel/next.js#13861] [Help]
I have re-produced that tree shaking is broken in my monorepo project. When importing anything from material-ui/mdi-material-ui/ or my core node_module using a named import , it causes all the lib to present in the chunks. When i check lighthouse in debug mode, i could see 5mb has beee downloaded. 

- https://github.com/VivekNeel/monorepo/blob/master/web/pages/index.jsx#L10
- https://github.com/VivekNeel/monorepo/blob/master/web/pages/index.jsx#L6
- https://github.com/VivekNeel/monorepo/blob/master/web/pages/index.jsx#L4

Please can anyone take a look at my code and see why is it not working?

**Comment by VivekNeel** [comment]
Because of this i noticed there is huge delay in fast refresh 8-10 sec, When using default export i dont have any problem.

**Reply by VivekNeel** [reply]
![image](https://user-images.githubusercontent.com/5962275/83967153-96882e00-a8dc-11ea-8791-757c519c1283.png)

**Comment by VivekNeel** [comment]
It's happening with just normal project set without using monorepo and also importing anything as named import from src and not from node_modules. For e.g am importing a named component `EditorFact` https://github.com/VivekNeel/Create-Test/blob/master/pages/index.js#L4 but when i checked the chunks it imported everything inside /src/create/index.js 
![image](https://user-images.githubusercontent.com/5962275/83967569-78bcc800-a8e0-11ea-8df3-787475757362.png)

CreateTerms component i did not import in pages/index but i still see it in chunk :(

**Reply by VivekNeel** [reply]
Anyone has any idea?

**Reply by FeelGoodInc** [reply]
Same trouble, but with semantic-ui-react. It's ok in production mode, but all semantic-ui included in development. So, I have 14mb page size (2.6 gziped) and fast-refresh isn't fast)) Have no idea, how to fix.

**Reply by VivekNeel** [reply]
@FeelGoodInc did you figure it out the fast refresh issue?

**Reply by FeelGoodInc** [reply]
@VivekNeel no
